temporalio/exceptions.py

```diff
@@ -23,7 +41,15 @@ def cause(self) -> Optional[BaseException]:
 
 
 class FailureError(TemporalError):
-    """Base for runtime failures during workflow/activity execution."""
+    """Base for runtime failures during workflow/activity execution.
+
+    This is extended by ``ApplicationError``, which can be raised in a Workflow to fail the Workflow Execution.
+    Workflow Execution Failures put the Workflow Execution into the "Failed" state and no more attempts will
+    be made in progressing this execution.
+
+    Any exception that does not extend this exception
+    is considered a Workflow Task Failure. These types of failures will cause the Workflow Task to be retried.
```

Comment:
It's actually more complicated than this since it is now configurable. We dedicate a section of our README to this at https://github.com/temporalio/sdk-python?tab=readme-ov-file#exceptions. Ideally we don't have to duplicate so much in here and there considering they may not say the same things, as is the case already here.